### PR TITLE
Feature/Checkpoint tests

### PR DESCRIPTION
**Pull request by kurisusnowdeng**
- Added unit tests for checkpoint save/load
- Updated checkpoint file extention to ignore in .gitignore

### CODE REVIEW

**Comment by feifeibear** [comment]
#256

File: tests/test_utils/test_checkpoint/test_checkpoint_3d.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import torch.nn as nn
+from colossalai.context.parallel_mode import ParallelMode
+from colossalai.core import global_context as gpc
+from colossalai.initialize import launch
+from colossalai.logging import disable_existing_loggers
+from colossalai.utils import free_port, get_current_device, is_using_pp
+from colossalai.utils.checkpointing import gather_pipeline_parallel_state_dict, load_checkpoint, save_checkpoint
+
+
+def build_pipeline(model):
```
Comment (feifeibear) [inline tests/test_utils/test_checkpoint/test_checkpoint_3d.py:20]: move this function to a common file.

File: tests/test_utils/test_checkpoint/test_checkpoint_2d.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    sd1 = m1.state_dict()
+    if gpc.get_global_rank() == 0:
+        print(f"Rank {gpc.get_global_rank()}:\n{pprint.pformat(sd1)}\n")
+    save_checkpoint("test.pt", 0, m1)
+
+    m2 = nn.Sequential(col_nn.Linear(4, 8), col_nn.Linear(8, 4))
+    if is_using_pp():
+        m2 = build_pipeline(m2)
+
+    load_checkpoint("test.pt", m2)
```
Comment (feifeibear) [inline tests/test_utils/test_checkpoint/test_checkpoint_2d.py:58]: the pt has the same name. use variable name please?
Rm the .pt after unittest?

Comment (kurisusnowdeng) [reply tests/test_utils/test_checkpoint/test_checkpoint_2d.py:58]: @feifeibear No problem.

File: tests/test_utils/test_checkpoint/test_checkpoint_3d.py (hunk 1 above)
Comment (kurisusnowdeng) [reply tests/test_utils/test_checkpoint/test_checkpoint_3d.py:20]: @feifeibear No problem.

**Review by feifeibear — APPROVED** [review]

**Pull request merged**